lib/mongo/uri/srv_protocol.rb

```diff
@@ -103,12 +103,23 @@ def parse_creds_hosts!(string)
         @password = parse_password!(creds)
       end
 
+      # Validates that a hostname fits the expected format.
+      #
+      # @param [ String ] hostname The hostname to validate.
+      #
+      # @raise [ Mongo::Error::InvalidURI ] If any of the following conditions are true:
+      #   * the hostname is empty
+      #   * the hostname contains either ',' or ':'
+      #   * the hostname begins with '.'
+      #   * the hostname contains consecutive '.' characters
```

Comment:
Please also state that the trailing dots are not allowed.

Reply:
That's incorrect; trailing dots are not forbidden by the spec, which is why there is a test that verifies it works